src/Array.hpp

```diff
@@ -439,7 +473,7 @@ class Array : public ArrayView< T,
    *        capacity() >= newCapacity.
    */
   void reserve( INDEX_TYPE const newCapacity )
-  { bufferManipulation::reserve( this->m_dataBuffer, this->size(), newCapacity ); }
+  { bufferManipulation::reserve( this->m_dataBuffer, this->size(), MemorySpace::host, newCapacity ); }
```

Comment:
shouldn't the space be propagated to the outer interface?

Reply:
I don't think so. At the moment the only way to allocate on device (without moving from host) is call `resizeWithoutInitializationOrDestruction` on an empty Array. If the Array already has an allocation (on host or device) it won't work. This is probably the majority use case, and if you want to resize the device only Array you can create a new device only Array and copy the values over. Doing a more general resize/reallocation on device without launching a kernel is tricky and not possible in all cases such as any time you have a default value. I'd certainly like to expand this capability and as soon as there's a use case it should be easy to add.

Doing the single parameter resize where the resize dimension is not the largest stride is serial (basically inserting new values in a bunch of places) and doing so on device would involve either creating a new device allocation each time or moving it back to host and doing the work there.